src/main/java/seedu/duke/command/timetable/TimeTableCommand.java

```diff
@@ -0,0 +1,108 @@
+package seedu.duke.command.timetable;
+
+import seedu.duke.command.Command;
+import seedu.duke.command.CommandResult;
+import seedu.duke.data.Module;
+import seedu.duke.data.ModuleManager;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import static seedu.duke.ui.TextUi.DIVIDER_LINE;
+
+public class TimeTableCommand extends Command {
+    public static final String COMMAND_WORD = "timetable";
+    public static final String FORMAT = COMMAND_WORD + " <opt>";
+    // todo add code into ModuleManager to get list of Module objects
+    private static ArrayList<Module> modCodeList = ModuleManager.getModCodeListAsModule();
+    public enum TypeOfTimetables {
+        DAY, WEEK
+    }
+    private final TypeOfTimetables flag;
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d-M-yyyy HHmm");
+    private DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HHmm");
+    private LocalDateTime now = LocalDateTime.now();
+
+    /**
+     * Option for timetable for the day or for the week
+     *
+     * @param flag Determines if the day or week timetable is shown, -d or -m
+     */
+    public TimeTableCommand (String flag) {
+        this.flag = TypeOfTimetables.valueOf(flag);
+    }
+
+    // Get a ArrayList of Module Objects for the day
+    public ArrayList<Module> getDayModCodeList() {
+        ArrayList<Module> dayModList = new ArrayList<>();
+        // Get the module object
+        for (Module modCode : modCodeList) {
+            // Same day as today
+            if (isSameDayAsToday(modCode)) {
+                dayModList.add(modCode);
+            }
+        }
+        return dayModList;
+    }
+
+    // Get a ArrayList of Module Objects for the day
+    public ArrayList<Module> getWeekModCodeList() {
+        ArrayList<Module> weekModList = new ArrayList<>();
+        // Get the module object
+        for (Module modCode : modCodeList) {
+            // Same day as today
+            if (isSameWeek(modCode)) {
+                weekModList.add(modCode);
+            }
+        }
+        return weekModList;
+    }
+
+    private boolean isSameDayAsToday(Module modCode) {
+        // todo Module toLocalDate function with a LocalDateTime
+        return now.toLocalDate().isEqual(modCode.toLocalDate());
+    }
+
+    private boolean isSameWeek(Module modCode) {
+        // todo Module toLocalDate function with a LocalDateTime
+        return (now.isBefore(modCode.toLocalDateTime()) && now.plusDays(6).isAfter(modCode.toLocalDateTime()));
+    }
+
+    public String dayTimeTable() {
+        ArrayList<Module> dayModList = getDayModCodeList();
```

Comment:
The logic in this method (and the next method below) will be moved to the new TimeTable class